src/core/Akka/Actor/ActorCell.cs

```diff
@@ -534,7 +537,8 @@ private Envelope SerializeAndDeserialize(Envelope envelope)
                 throw new SerializationException($"Failed to serialize and deserialize payload object [{unwrapped.GetType()}]. Envelope: [{envelope}], Actor type: [{Actor.GetType()}]", e);
             }
 
-            if (deadLetter != null)
+            // special case handling for DeadLetters
+            if (envelope.Message is DeadLetter deadLetter)
```

Comment:
It seems like we're missing a round of unwrap here, this code will return an `Envelope` that wraps a `DeadLetter` that wraps a `DeadLetter`.
The `DeadLetter` message still needs to be deserialized like it was before and then re-wrapped inside a new `DeadLetter` message.

Reply:
Good point, I can use the `Unwrapped` utility from earlier to recursively unwrap it - let me look into that. Currently doing a full test suite run locally so I can fix some of the outstanding .NET 8 errors in the suite.